src/ServiceProvider.php

```diff
@@ -55,15 +55,16 @@ public function getPugEngine(): Pug
         $config = $this->getConfig();
 
         if (!isset($config['extensions']) && $this->app['view']) {
-            $extensions = array_keys(array_filter($this->app['view']->getExtensions(), function ($engine) {
-                $engines = explode('.', $engine);
+            $extensions = $this->app['view']->getExtensions();
+            $filteredExtensions = [];
 
-                return in_array('pug', $engines);
-            }));
+            foreach ($extensions as $key => $engine) {
+                if (strpos($engine, 'pug') !== false) {
```

Comment:
This is not equivalent as it searches for any pug sub-string not delimited by `.` separator.